vars/sconsBuild.groovy

```diff
@@ -2,49 +2,201 @@
+   *  symlinked from the target prefix.
+   * config['update_prereq'] Setting for --update-prereq.  Default 'all'.
+   * config['USE_INSTALLED'] setting for USE_INSTALLED.  Default 'yes'.
+   *  If false, a failure of the scons commands will cause this step to fail.
+   */
+
+    /* If we have to tamper with the checkout, we also need to remove
+     * the potential tampering before the scm operation.
+     */
+    if (config['scons_local_replace'] && config['target']) {
+       sh "rm -rf \"\${WORKSPACE}/${config['target']}/scons_local\""
+    }
+
+    scm_config = [withSubmodules: true]
+    if (config['scm']) {
+        scm_config = config['scm']
+        if (config['target'] && !scm_config['checkoutDir']) {
+            scm_config['checkoutDir'] = config['target']
+        }
+    }
+    checkoutScm(scm_config)
+    if (env.DAOS_JENKINS_NOTIFY_STATUS != null) {
+        githubNotify credentialsId: 'daos-jenkins-commit-status',
+                     description: env.STAGE_NAME,
+                     context: "build" + "/" + env.STAGE_NAME,
+                     status: "PENDING"
+    }
+
+    if (config['scons_local_replace'] && config['target']) {
+       // replace the scons_local directory in config['target']
+       // with a link to the scons_local directory.
+       sh """rm -rf "\${WORKSPACE}/${config['target']}/scons_local"
+             ln -s "${WORKSPACE}/scons_local" \
+               "\${WORKSPACE}/${config['target']}/scons_local"
+             """
+    }
+
+    set_cwd = ''
+    if (config['directory']) {
+        set_cwd = "cd ${config['directory']}\n"
+    } else if (config['target']) {
+        set_cwd = "cd ${config['target']}\n"
+    }
+
+    scons_args = ''
+    sconstruct = ''
+    if (config['sconstruct']) {
+        sconstruct = " --sconstruct=${config['sconstruct']}"
+        scons_args += sconstruct
+    }
+    if (config['update_prereq']) {
+        scons_args += " --update-prereq=${config['update_prereq']}"
+    } else {
+        scons_args += ' --update-prereq=all'
+    }
+    if (config['build_deps']) {
+        scons_args += " --build-deps=${config['build_deps']}"
+    } else {
+        scons_args += ' --build-deps=yes'
+    }
+    if (!config['no_install']) {
+        scons_args += ' install'
+    }
+    if (config['USE_INSTALLED']) {
+        scons_args += " USE_INSTALLED=${config['USE_INSTALLED']}"
+    } else {
+        scons_args += ' USE_INSTALLED=all'
+    }
+    if (config['SRC_PREFIX']) {
+        scons_args += " SRC_PREFIX=${config['SRC_PREFIX']}"
+    } else if (config['target']) {
+        scons_args += " SRC_PREFIX=\${WORKSPACE}}"
+    }
+    if (config['REQUIRES']) {
+        scons_args += " REQUIRES=${config['REQUIRES']}"
+    } else if (config['target']) {
+        scons_args += " REQUIRES=${config['target']}"
+    }
+    //scons -c is not perfect so get out the big hammer
+    clean_files = "_build.external* install build"
+    if (config['clean']) {
+        clean_files = config['clean']
```

Comment:
This implementation of `config['clean']` is conflicting with an existing use of `config['clean']` in [sconsBuild-clean](https://github.com/daos-stack/pipeline-lib/commit/62dd57e7becdd6845cf7a0d65ecfac09006a9225)[1] which existing pipelines are using.  To be backwards compatible with that existing use and not need a flag-day change of existing pipelines:

```suggestion
        clean_files = config['clean'] + 'install build'
```

[1] I didn't put that into a PR and land it yet as it was planned to be obsolete by harmonzing the use (or lack thereof as it now appears it will be) of the _arch_ in the build across daos, iof and cart.

Reply:
Done.